sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -1139,8 +1154,19 @@ private ProgramState VisitOpAssignment(AssignmentExpressionSyntax assignment, Pr
 
         private ProgramState VisitSimpleAssignment(AssignmentExpressionSyntax assignment, ProgramState programState)
         {
+            if (DeclarationExpressionSyntaxWrapper.IsInstance(assignment.Left))
```

Comment:
are there other places where `AssignmentExpressionSyntax` gets processed and a similar logic should be emplyed?

Reply:
A `AssignmentExpressionSyntax` is used in the following cases:
```
SyntaxKind.SubtractAssignmentExpression:
SyntaxKind.AddAssignmentExpression:
SyntaxKind.DivideAssignmentExpression:
SyntaxKind.MultiplyAssignmentExpression:
SyntaxKind.ModuloAssignmentExpression:
SyntaxKind.LeftShiftAssignmentExpression:
SyntaxKind.RightShiftAssignmentExpression:
SyntaxKind.OrAssignmentExpression
SyntaxKind.AndAssignmentExpression
SyntaxKind.ExclusiveOrAssignmentExpression
```
but a tuple cannot be part of these.